src/Avalonia.Controls/TopLevel.cs

```diff
@@ -232,6 +235,18 @@ protected bool AutoSizing
         }
 
         /// <inheritdoc/>
+        IRenderTarget IRenderRoot.CreateRenderTarget()
+        {
+            return _renderInterface.CreateRenderer(PlatformImpl.Handle);
```

Comment:
I think `ITopLevelImpl` implementation should obtain `IRenderTarget` from rendering subsystem (using IPlatformHandle *or* bitmap-backed) and manage it's lifetime by itself.